### ISSUE TITLE
Ubuntu 18.04 issue just installing

### ISSUE THREAD

**Issue by cmseal** [voxpupuli/puppet-php#475]
Puppet agent 5.5.6
Just want to install PHP...

Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install python-software-properties' returned 100: Reading package lists...
Building dependency tree...
Reading state information...
Package python-software-properties is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  software-properties-common

**Comment by bastelfreak** [comment]
Hi @cmseal. Can you please provide a few more information? How did you call this module? Which version do you use? Which version of the apt module?

**Comment by cmseal** [comment]
Hi @bastelfreak, sorry here's some more info... if any other info will help, let me know.

Clean 18.04.1 LTS server with apt 1.6.3ubuntu0.1 (amd64), and module is latest pulled from master, as a submodule on my puppet master. Called in manifest with "include ::php" with hiera config:

php::extensions:
    mysql: {}
    mbstring: {}
    mcrypt: {}
    gd: {}

Full error is:
Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install python-software-properties' returned 100: Reading package lists...
Building dependency tree...
Reading state information...
Package python-software-properties is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  software-properties-common

E: Package 'python-software-properties' has no installation candidate
Error: /Stage[main]/Php::Repo::Ubuntu/Apt::Ppa[ppa:ondrej/php]/Package[python-software-properties]/ensure: change from 'purged' to 'present' failed: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install python-software-properties' returned 100: Reading package lists...
Building dependency tree...
Reading state information...
Package python-software-properties is not available, but is referred to by another package.
This may mean that the package is missing, has been obsoleted, or
is only available from another source
However the following packages replace it:
  software-properties-common

E: Package 'python-software-properties' has no installation candidate
Notice: /Stage[main]/Php::Repo::Ubuntu/Apt::Ppa[ppa:ondrej/php]/Exec[add-apt-repository-ppa:ondrej/php]: Dependency Package[python-software-properties] has failures: true

**Comment by bastelfreak** [comment]
mcrypt isn't a packaged module on Ubutnu 18.04 because it got deprecated for PHP 7.2. Puppet output:

```
  Error: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install php7.2-mcrypt' returned 100: Reading package lists...
  Building dependency tree...
  Reading state information...
  E: Unable to locate package php7.2-mcrypt
  E: Couldn't find any package by glob 'php7.2-mcrypt'
  E: Couldn't find any package by regex 'php7.2-mcrypt'
  Error: /Stage[main]/Php/Php::Extension[mcrypt]/Php::Extension::Install[mcrypt]/Package[php7.2-mcrypt]/ensure: change from 'purged' to 'present' failed: Execution of '/usr/bin/apt-get -q -y -o DPkg::Options::=--force-confold install php7.2-mcrypt' returned 100: Reading package lists...
  Building dependency tree...
  Reading state information...
  E: Unable to locate package php7.2-mcrypt
  E: Couldn't find any package by glob 'php7.2-mcrypt'
  E: Couldn't find any package by regex 'php7.2-mcrypt'
```

* It isn't on the [official mirrors](https://packages.ubuntu.com/search?keywords=mcrypt&searchon=names&suite=bionic&section=all)
* [Deprecation notice](https://secure.php.net/manual/en/migration71.deprecated.php)

**Comment by bastelfreak** [comment]
I added some tests in https://github.com/voxpupuli/puppet-php/pull/476. It seems that ```software-properties-common``` is pulled in via https://github.com/voxpupuli/puppet-php/pull/419. It manages the needed package to add ppa:

https://github.com/puppetlabs/puppetlabs-apt/blob/484d13b8f166353cc1a18ad03e1aedee08e5753b/manifests/ppa.pp#L59-L65

On ```Ubuntu``` this is always:

https://github.com/puppetlabs/puppetlabs-apt/blob/484d13b8f166353cc1a18ad03e1aedee08e5753b/manifests/params.pp#L90-L98

**Comment by bastelfreak** [comment]
I added a test that also installs extensions. Works fine on ubuntu: https://github.com/voxpupuli/puppet-php/pull/476

**Comment by dhollinger** [comment]
This issue was solved with #482
